internal/gather/cni.go

```diff
@@ -135,26 +134,49 @@ func gatherOVNResources(info *Info, networkPlugin string) {
 		return
 	}
 
-	// we check two different labels because OpenShift deploys with a different
-	// label compared to ovn-kubernetes upstream
-	ovnMasterpods, err := findPods(info.ClientProducer.ForKubernetes(), ovnMasterPodLabelOCP)
-	if err != nil || ovnMasterpods == nil || len(ovnMasterpods.Items) == 0 {
-		ovnMasterpods, err = findPods(info.ClientProducer.ForKubernetes(), ovnMasterPodLabelGeneric)
-		if err != nil {
-			info.Status.Failure("Failed to gather any OVN master ovnMasterpods: " + err.Error())
-		} else if ovnMasterpods == nil || len(ovnMasterpods.Items) == 0 {
-			info.Status.Failure("Failed to find any OVN master ovnMasterpods")
+	ovnKubePod := getOVNCmdsPod(info)
+	if ovnKubePod != nil {
+		info.Status.Success("Gathering OVN data from OVNkube pod %q", ovnKubePod.Name)
+
+		for name, command := range ovnCmds {
+			logCmdOutput(info, ovnKubePod, command, name, false)
 		}
 	}
 
-	info.Status.Success("Gathering OVN data from master pod %q", ovnMasterpods.Items[0].Name)
+	gatherGatewayRoutes(info)
+	gatherNonGatewayRoutes(info)
+}
+
+func getOVNCmdsPod(info *Info) *v1.Pod {
+	/*
+			We use three different labels for different OVNKubernetes deployments
+		      * ovnMasterPodLabelOCP     - Non IC OpenShift
+		      * ovnMasterPodLabelGeneric - Non IC upstreamovn-kubernetes
+		      * ovnKubePodLabel          - IC Deployments, same for upstream and OpenShift
+	*/
+	ovnCmdPods, _ := findPods(info.ClientProducer.ForKubernetes(), ovnMasterPodLabelOCP)
 
-	for name, command := range ovnCmds {
-		logCmdOutput(info, &ovnMasterpods.Items[0], command, name, false)
+	if ovnCmdPods != nil && len(ovnCmdPods.Items) > 0 {
```

Comment:
```suggestion
	if len(ovnCmdPods.Items) > 0 {
```

Reply:
Was in two minds for this change, followed convention from existing code. But will go with this coz it is better and simpler.